docs/getting-started/installation/linux.md

```diff
@@ -4,76 +4,50 @@ sidebar_position: 3
-  the guide will also explain how to install NPM. This way you'll have Node, NPM, and then
-  you can run `npm install -g yarn` to install Yarn.
-
-### PostgreSQL
-
-Forem requires PostgreSQL version 11 or higher.
-
-1. Run
-   `sudo apt update && sudo apt install postgresql postgresql-contrib libpq-dev`.
-1. To test the installation you can run `sudo -u postgres psql` which should
-   open a PostgreSQL prompt. Exit the prompt by running `\q` then run
-   `sudo -u postgres createuser -s $YOUR_USERNAME` where `$YOUR_USERNAME` is the
-   username you are currently logged in as.
-
-There are more than one ways to setup PostgreSQL. For additional configuration,
-check out our [PostgreSQL setup guide](postgresql.md) or the official
-[PostgreSQL](https://www.postgresql.org/) site for further information.
-
-### ImageMagick
-
-Forem uses [ImageMagick](https://imagemagick.org/) to manipulate images on
-upload.
-
-Please refer to ImageMagick's
-[instructions](https://imagemagick.org/script/download.php) on how to install
-it.
-
-### Redis
-
-Forem requires Redis version 6.0 or higher.
-
-We recommend following Digital Ocean's extensive guides
-[How to Install and Secure Redis](https://www.digitalocean.com/community/tutorial_collections/how-to-install-and-secure-redis)
-to setup Redis.
-
-## Installing Forem
+**Installing Forem on Bare-Metal Linux**
+
+> As the title implies, this spins up a Forem stack for local development with no use of containerization, virtualization, etc., which comes with tradeoffs: standing the stack up will almost invariably be a slower process as various tools build from source, but attaching debuggers and the like to the resulting Forem application may be easier. If using containers through Docker or Podman isn't a deal-breaker for your usecase, [consider those docs instead](containers.md) for faster stand-up and teardown, and to leave less cruft behind when you're done working.
+
+_For maintainer sanity reasons, this opinionated and curated instruction set assumes you're cool with the use of user-local installations of all dependencies using [rtx](https://github.com/jdx/rtx) as a project-scoped package manager. If `rtx` is for any reason a non-starter for you, you'll want to use your package management solutions of choice to install the tools found in `.rtx.toml`, `.ruby-version`, and `.nvmrc`, at the appropriate versions, and should be comfortable sorting out the dependencies thereof on your own. The instructions were written for Debian 12 (Bookworm), but should be directly usable on Ubuntu, Mint, and other Debian derivatives, and should be adaptable for other distributions._
+
+## Install System-Level Dependencies
+
+- [rtx](https://github.com/jdx/rtx)
+- [ImageMagick](https://imagemagick.org/), used to manipulate images on upload. Currently, there's not a reliable way to manage this with `rtx`, so you'll need to use your distribution package manager
```

Comment:
This is a minor thing — and maybe I am just not the best reviewer here, as I'm on an M1 Mac — but I think this might be a step in the wrong direction, from a documentation perspective. The previous version of this linked to `ImageMagick`'s site (`https://imagemagick.org/script/download.php`) which has _relatively comprehensive_ instructions on installing. 

To be clear, they are not _great_ — they make some assumptions about the reader's ability to install the binary — and the system package manager _might_ be better advice, but I feel like it's better to link to something _concrete_ and _authoritative_ than _nothing_.

In general, I'm not _totally_ on-board with the push of `rtx` here. With the caveat, again, that I'm on a Mac, I've had issues with `rtx` not being _universally_ better than other management tools (`rbenv` handles M1 compilation and `openssl` shenanigans better than `rtx` does, as of writing). 

But, more so, a big selling point of `rtx` and `asdf`-type "multi-tool managers" is that they operate well with "legacy version files" — `.node-version`, `.ruby-version`, etc. On a technical level, there's no _requirement_ that anyone use `rtx`, it's just the "tool manager" we currently like best.

I think it's fine to point folks at _our recommendation_ — we were doing that before with `rbenv`, and I think it's _fine_ to switch that to `rtx` for Linux folks (I, personally, still think `rbenv` is better for M1 Mac folks, at least, but I am fully neutral as to what we recommend on Linux), I think it's a step backwards to have these instructions written with a _specific_ tool as their focus. 

If nothing else, I really don't think we need to repeat `rtx`'s instructions for setting up tool management.

Reply:
Gonna tackle this in chunks (and if I don't get around to responding to all of it in this pass, I'll respond to the rest tomorrow). Firstly though, thank you for the write-up here and for all the thought put into it, I really appreciate it!

I fell back to system package manager for ImageMagick only because [the asdf plugin for managing source-based IM installs](https://github.com/mangalakader/asdf-imagemagick) doesn't support MacOS at all (ARM or Intel), otherwise there's a Linux-friendly solution there. I'm open to the idea of forking and fixing that plugin to support MacOS, and thus removing the system-level dependency entirely here, though I'm less familiar on whether our app code is using a Gem to talk to ImageMagick or if it's calling out to the system using `$PATH`. The latter is a great fit for rtx/asdf-based setups; the former I'm not entirely certain about ("in theory" the Gem "just" builds against whatever libraries are in the include paths that `pkg-config`, `gcc`, etc. find, but the existence of... the entire ecosystem worth of C/C++ package managers and their tangled webs of complexity tells us that "just" is a loaded word here).

---

For `rtx` as the "true path", I'm still poking at the MacOS edges (see PR description) here so I haven't found every wart, but given that `rtx` is using `ruby-build` under the hood here the same as `rbenv`, I'm somewhat surprised to hear you had unique challenges there. I'll hold too many further MacOS-specific thoughts until I've finished the port-over here.

Anyway, I wanted to use `rtx` as the "true path" here (for both Linux and MacOS) because of two main factors: 1, that it can pin exact versions (and build them from source if needed) in a way that few cross-platform package managers can (Nix can do it, but speaking from experience rolling out Nix to an org before: no, absolutely not, I do not want to maintain that as the "primary" path), and 2, because it can handle (via `asdf` plugins) just about any dev tool under the sun in the same general way, regardless of underlying OS, and it does so performantly. I retained a note in the docs about how one is welcome to *not* use `rtx`, but I generally prefer (so this is subjective and somewhat philosophical) having Exactly One Supported Way To Do It and letting the user explore on their own terms if they for some reason disagree with the choices or can't use the given tool. To me, the existing docs read like a bit of a "choose your own adventure" novel, whereas for expedient external contributor onboarding, I think I prefer "here's the battle-tested path that works in under 20 minutes, if you want to do something else, none of this stuff is proprietary or hidden behind too many layers of paint, go for it".

This philosophy is also the core of why I copied out `rtx`'s installation instructions into this guide, so I'll reuse that "all in one place, all as one tested set of instructions" rationale here. I should also note that I do more or less the same for Homebrew in the draft of the MacOS docs I have sitting around, so this is probably something we should figure out some agreed philosophy on before long; I'd like to have a common voice between both documents if I can.

---

Anyway! None of this is set in stone, I appreciate having someone who *isn't* a Linux daily driver (Mac and I both are) to bounce ideas off of here and to remind me of blind spots, so consider this reply more as providing backstory and context (and asking for more conversation on some areas) than it is necessarily "defending" one way or the other quite yet :smiley:

Reply:
@jaw6 could you take a look at @klardotsh comment Friday? Thank you so much!

Reply:
@mirie @klardotsh 

For me, I mean, evaluating this kind of change is tricky, because I have not actually _used_ the earlier version of this document to stand up a Forem instance on any Linux. And - I imagine - you actually have that experience! If this document would have been more helpful in that effort, I don't want to stand in the way of improvement.

For me, for what it's worth, I _don't really_ think the issue is `rtx`, or version managers generally, _per se_, but that's the concrete detail I was tripping on earlier.

I think the previous version of this having linked to `ImageMagick`'s install documentation _seems_ like it would be a more useful resource than it would be to _remove that link_. Describing that there isn't a reliable way to manage this with `rtx` doesn't seem like a net improvement over linking to deeper instructions. Since we've disclaimed this with "instructions were written for Debian 12", maybe we could actually include the details on how to install `imagemagick` on debian? (Since those should more-or-less work for debian-derived distributions, I think?)

Again, for what it's worth, just, like, as a reader, the version of this doc on `main` is _organized_ and _coherent_ in a way that the proposed change is not. It starts with installing pre-requisites, it sometimes recommends version managers, it often links to detailed guides for more details. 

As a reader, the proposed change is hard to follow. `rtx` is _mentioned_ at the top of "Install System-Level Dependencies", then followed by a bullet-list of somehow vague dependencies — lacking installation instructions or specific version details — then a code block that details how to install `rtx`. (As I reader, I would have expected this code block to actually install the preceding list of dependencies.) Later, under "Install Application-Level Dependencies", there's discussion and links for alternatives to Postgres and Redis (this strangely being the first time redis is mentioned in the document, and the first time we've mentioned Postgres directly, rather its "various transitive dependencies").

Maybe we could try to preserve some of the original document's structural organization, even if we are shifting about things like `rtx`. That could be something like:

> * maintainer sanity disclaimer
> * how to install rtx
> * how to install dependencies with rtx
> * other dependencies not managed by rtx (imagemagick, say)
> * how to install forem (yarn install, etc.)
> * Possible error messages (this _seems_ useful, is it relevant now?)

(I'm not sure why `rtx` wasn't able to install ruby, it looked like it choked on `openssl`, which [has happened to me a lot](https://forem.team/jaw6/local-development-misadventures-5eh6) and was more-or-less resolved when homebrew and I got out of `ruby-build`'s way. I didn't dig deeper, I did not realize the ruby plugin was based on `ruby-build`, so that discrepancy is _interesting_, but I can look into it, at some point. It seems like it might be useful to figure that out.)